projects/core/src/cart/store/effects/multi-cart.effect.ts

```diff
@@ -52,34 +53,38 @@ export class MultiCartEffects {
   );
 
   @Effect()
-  setLoading$: Observable<CartActions.SetCartLoading> = this.actions$.pipe(
+  removeCart$: Observable<CartActions.RemoveCart> = this.actions$.pipe(
+    ofType(DeprecatedCartActions.DELETE_CART),
+    map((action: DeprecatedCartActions.DeleteCart) => action.payload),
+    map(payload => new CartActions.RemoveCart(payload.cartId))
+  );
+
+  @Effect()
+  processesIncrementAction$: Observable<
+    CartActions.CartProcessesIncrementAction
+  > = this.actions$.pipe(
     ofType(
-      DeprecatedCartActions.MERGE_CART,
       CartActions.CART_ADD_ENTRY,
       CartActions.CART_UPDATE_ENTRY,
-      CartActions.CART_REMOVE_ENTRY
+      CartActions.CART_REMOVE_ENTRY,
+      DeprecatedCartActions.ADD_EMAIL_TO_CART,
+      CheckoutActions.CLEAR_CHECKOUT_DELIVERY_MODE,
+      CartActions.CART_ADD_VOUCHER,
+      CartActions.CART_REMOVE_VOUCHER
     ),
     map(
       (
         action:
-          | CartActions.MergeCart
           | CartActions.CartAddEntry
           | CartActions.CartUpdateEntry
           | CartActions.CartRemoveEntry
+          | DeprecatedCartActions.AddEmailToCart
+          | CheckoutActions.ClearCheckoutDeliveryMode
+          | CartActions.CartAddVoucher
+          | CartActions.CartRemoveVoucher
       ) => action.payload
     ),
-    mergeMap(payload => [
-      new CartActions.SetCartLoading({
-        cartId: payload.cartId,
-      }),
-    ])
-  );
-
-  @Effect()
-  removeCart$: Observable<CartActions.RemoveCart> = this.actions$.pipe(
-    ofType(DeprecatedCartActions.DELETE_CART),
-    map((action: DeprecatedCartActions.DeleteCart) => action.payload),
-    mergeMap(payload => [new CartActions.RemoveCart(payload.cartId)])
+    map(payload => new CartActions.CartProcessesIncrementAction(payload.cartId))
```

Comment:
can we add a comment /ticket regarding the optimalization of Incrementing/Decrementing in 2.0?

Reply:
Added here: https://github.com/SAP/cloud-commerce-spartacus-storefront/issues/5234